thrift/thrift0.13/src/main/java/com/linecorp/armeria/internal/server/thrift/ThriftDocServicePlugin.java

```diff
@@ -199,7 +202,12 @@ private MethodInfo newMethodInfo(Method method, Iterable<EndpointInfo> endpoints
         }
         final ClassLoader classLoader = serviceClass.getClassLoader();
 
-        final String argsClassName = serviceName + '$' + methodName + "_args";
+        // Need get the function name in thrift proto,
+        // otherwise the argsClassName maybe wrong when use fullcamel compile option
+        final String thriftFunctionName = Optional.ofNullable(entry.thriftServiceEntry)
```

Comment:
Since `entry.thriftServiceEntry` is always non-null. 

```suggestion
        final String thriftFunctionName = Optional.of(entry.thriftServiceEntry)
```

Also I'm not strong on this, but what do you think of not using `Optional` since it just allocates an extra object with no real gain here? (It's acting as a if-else).  Ditto for the other usage.

Reply:
like this ?
```
        String thriftFunctionName = entry.thriftServiceEntry.functionName(methodName);
        if (thriftFunctionName == null) {
            thriftFunctionName = methodName;
        }
```
I'm not sure too, this way seems better to understand than optional.

Reply:
> Since `entry.thriftServiceEntry` is always non-null.
> 
> Also I'm not strong on this, but what do you think of not using `Optional` since it just allocates an extra object with no real gain here? (It's acting as a if-else). Ditto for the other usage.

Dig  code for a while, many tests use  `new EntryBuilder(HelloService.class)`  style to create builder directly, if I use make the `thriftServiceEntry` not null and final in the `Entry`, the creation of `thriftServiceEntry` in tests will be difficult to handle. 
So the modification could be very little  if we just let thriftServiceEntry nullable.
@jrhee17

Reply:
> So the modification could be very little if we just let thriftServiceEntry nullable.

I didn't realize it could be `@Nullable`. If it is, then can we add a `@Nullable` annotation to `ThriftDocService.Entry.thriftServiceEntry`?

Reply:
👌🏻